govcd/vapp_vm_test.go

```diff
@@ -92,3 +92,33 @@ func propertyTester(vcd *TestVCD, check *C, object productSectionListGetSetter)
 	check.Assert(gotproductSection.ProductSection.Property, DeepEquals, productSection.ProductSection.Property)
 	check.Assert(getproductSection, DeepEquals, gotproductSection)
 }
+
+// guestPropertyGetSetter interface is used for covering tests
+type getGuestCustomizationSectionGetSetter interface {
+	GetGuestCustomizationSection() (*types.GuestCustomizationSection, error)
+}
+
+// guestCustomizationPropertyTester is a guest customization property get and setter accepting guestPropertyGetSetter interface for trying
+// out settings on all objects implementing such interface
+func guestCustomizationPropertyTester(vcd *TestVCD, check *C, object getGuestCustomizationSectionGetSetter) {
+	guestCustomizationSection, err := object.GetGuestCustomizationSection()
+	check.Assert(err, IsNil)
+
+	// Check that values were set from API
+	check.Assert(guestCustomizationSection, NotNil)
+
+	check.Assert(guestCustomizationSection.Enabled, Equals, true)
+	check.Assert(guestCustomizationSection.JoinDomainEnabled, Equals, false)
+	check.Assert(guestCustomizationSection.UseOrgSettings, Equals, false)
+	check.Assert(guestCustomizationSection.DomainUserName, Equals, "")
+	check.Assert(guestCustomizationSection.DomainName, Equals, "")
+	check.Assert(guestCustomizationSection.DomainUserPassword, Equals, "")
+	check.Assert(guestCustomizationSection.AdminPasswordEnabled, Equals, true)
+	check.Assert(guestCustomizationSection.AdminPasswordAuto, Equals, true)
+	check.Assert(guestCustomizationSection.AdminPassword, Equals, "")
+	check.Assert(guestCustomizationSection.AdminAutoLogonCount, Equals, 0)
+	check.Assert(guestCustomizationSection.AdminAutoLogonEnabled, Equals, false)
+	check.Assert(guestCustomizationSection.ResetPasswordRequired, Equals, false)
+	check.Assert(guestCustomizationSection.CustomizationScript, Equals, "")
+	check.Assert(guestCustomizationSection.ComputerName, Equals, "PhotonOS-001")
```

Comment:
Is it always Photon? Can't we change the OVA to something else in the test configuration file?

Reply:
Changed